pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/extension/PythonDetails.java

```diff
@@ -133,6 +131,17 @@ public void setSystemPythonInterpreter(String path) {
     }
 
     public PythonVersion getPythonVersion() {
+        findPythonWhenAbsent();
         return pythonVersion;
     }
+
+    private void findPythonWhenAbsent() {
+        if (pythonInterpreter == null) {
+            File python = ExecutablePathUtils.getExecutable(searchPath, "python");
```

Comment:
This is intended to be a fallback. If you specify a version to use, or a specific interpreter it will use that (because pythonInterpreter will be non null).